content/blog/2023/08/02/2023-08-23-gitlab-plugin-modernization-report.adoc

```diff
@@ -0,0 +1,89 @@
+
+The migration was performed in the following 2 coding phases:
+
+---
+
+== Phase 1
+
+In this phase, the following tasks were completed:
+
+* Migration of GitLab Plugin from RESTEasy to GitLab4J-API via the GitLab-API plugin, without affecting the webhook functionality of the plugin. (link:[PR])
+* Interactive testing of partially migrated plugin.
+
+The achievenments of this phase were demonstrated in the (link:https://youtu.be/W4eSVCTmqb8[Midterm Presentation])
+
+---
+
+== Phase 2
+
+In this phase, the following tasks were completed:
+* Adapting webhooks implementation to GitLab4J-API.(link:https://github.com/jenkinsci/gitlab-plugin/pull/1553[PR])
+* Migrating the Proxy Settings from RESTEasy to GitLab4J-API.(link:[PR])
+* Partially adapting the Unit and Integration tests.
+* Improving the documentation of the plugin.(link:[PR])
+* Interactive testing of the fully migrated plugin.
+
+The results of this phase will be showcased in in the Final Presentation in the future.
+---
+
+== Future Improvements
+
+- As GitLab has discontinued the GitLab API V3, the plugin should only support GitLab API V4 and above. Thus the plugin will be updated to support GitLab API V4 and above only.
+- The plugin lacks the test coverage, thus the plugin's Unit, Integrations as well as Docker-based Tests will have to be updated.
+- GitLab 16.0 has introduced some breaking changes which will have to be handled in the plugin.
+
+---
+
+== Acknowledgments
+
+* My gratitude knows no bounds that I got the opportunity to contribute to such impactful project under Google Summer of Code with such kind mentors. I would like to thank my mentors, link:/blog/authors/krisstern/[Kris Stern] and link:/blog/authors/MarkEWaite/[Mark Waite], for their constant support and guidance throughout the project. They put in a lot of effort in interactively testing the migrated plugin and providing me with valuable feedback. They never let me be stuck in any difficulty for too long and always helped me in finding the solution to the problem. I would also like to thank link:/blog/authors/basil/[Basil Crow] for the guidance he provided me with during the initial phase of the project, which helped the team shape the project.
+
+* Also, thanks to the org admins, link:/blog/authors/jmmeessen[Jean-Marc Meessen], link:/blog/authors/alyssat[Alyssa Tong], link:/blog/authors/krisstern/[Kris Stern], and link:/blog/authors/gounthar/[Bruno Verachten], for always keeping me and other contributors on our toes and assisting us with any blockers and concerns by organizing weekly stand-up calls and personal 1-to-1 meets every phase.
+
+---
+
+== Useful Links
+
+- link:https://plugins.jenkins.io/gitlab-plugin/[GitLab Plugin]
+- link:https://github.com/jenkinsci/gitlab-plugin[Plugin's GitHub Repository]   
+- Use the link:https://matrix.to/#/#jenkinsci_gitlab-plugin:gitter.im[Gitter channel] or link:https://community.jenkins.io[community.jenkins.io] in case you have any question(s) or feedback.
```

Comment:
```suggestion
- Use the link:++https://matrix.to/#/#jenkinsci_gitlab-plugin:gitter.im++[Gitter channel] or link:https://community.jenkins.io[community.jenkins.io] in case you have any question(s) or feedback.
```